### ISSUE TITLE
opentelemetry-collector: add option to generate ServiceMonitor

### ISSUE THREAD

**Issue by kthompson23** [honeycombio/helm-charts#14]
The opentelemetry-collector and this chart exposes Prometheus metrics on port 8888. Many charts add an option to generate a ServiceMonitor for users of the Prometheus Operator. We are using Honeycomb to observe our services but Prometheus for the infrastructure.

I think the kured [chart](https://github.com/weaveworks/kured/blob/master/charts/kured/templates/servicemonitor.yaml) has a nice api for generating the ServiceMonitor.

**Comment by puckpuck** [comment]
I agree this will be good, but before we add more ports to the deployment, we want to make all defined ports become properties of the chart itself, so this will be part of a larger undertaking.

**Comment by melissafritcher** [comment]
@puckpuck Are y'all accepting contributions? I could make all the defined ports into chart properties.

**Comment by puckpuck** [comment]
Before we add more ports (we are missing a few), I want to make sure we have existing ports opened and defined properly in config, deployment, and service. By default this should cover all Zipkin, Jaeger, and OTLP ports.  Jaeger in particular has a few ports that are not covered.

PRs are welcome.

**Comment by matoszz** [comment]
hey @puckpuck I also was also trying to PR the port mappings but you have to open up contributions on the repo, otherwise we can't create branches / open PR's

**Comment by puckpuck** [comment]
@matoszz you should be able to fork the repo and submit a PR from your fork. We only allow Honeycomb employees to add branches to this repo.

**Comment by puckpuck** [comment]
We recently released #30 which allows you to quickly configure ports like 8888 for metrics. It is not a default opened port, but if you feel the chart's default should have this enabled please file another issue and we will consider it. 

To enable the metrics port, you would add this to your custom values file passed into the chart:
```
ports:
  metrics:
    enabled: true
    containerPort: 8888
    servicePort: 8888
    protocol: TCP
```
